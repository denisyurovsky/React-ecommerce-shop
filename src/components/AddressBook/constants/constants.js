export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
