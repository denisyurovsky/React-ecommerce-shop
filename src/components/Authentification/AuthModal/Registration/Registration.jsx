import { Box, Button } from '@mui/material/';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import {
  MIN_PASSWORD_LENGTH,
  authStatus,
} from '../../../../constants/authConstants.js';
import checkEmailValidity from '../../../../helpers/checkEmailValidity.js';
import checkPasswordValidity from '../../../../helpers/checkPasswordValidity.js';
import LoadingButton from '../../../ui-kit/buttons/LoadingButton';
import { StandartedTextField } from '../../../ui-kit/StandartedTextField/StandartedTextField.jsx';
import PasswordForm from '../PasswordForm';
import useErrorHandler from '../useErrorHandler.js';

const { PENDING, FULFILLED, REJECTED, IDLE } = authStatus;

const Registration = ({ sendForm, errorMessage, status }) => {
  const isLoading = status === PENDING;
  const isError = status === REJECTED;
  const { isError: isServerError, resetError } = useErrorHandler(isError);

  const [values, setValues] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
  });

  const [confirmPassword, setConfirmPassword] = useState('');
  const [showPassword, setShowPassword] = useState(false);

  const [checkValues, setCheckValues] = useState({
    isEmailValid: true,
    isPasswordValid: true,
    isConfirmPasswordValid: true,
  });

  const [isFinished, setIsFinished] = useState(false);

  const checkName = (name) => name.length > 0;
  const checkEmail = (email) => email.match(/@/);
  const checkPassword = (password) => password.length > MIN_PASSWORD_LENGTH;
  const checkMatch = (str1, str2) => str1 === str2;
  const checkIsFinished = ({
    firstName,
    lastName,
    email,
    password,
    confirmPassword,
  }) =>
    checkName(firstName) &&
    checkName(lastName) &&
    checkEmail(email) &&
    checkPassword(password) &&
    checkPassword(confirmPassword);

  const checkValidForm = ({ email, password, confirmPassword }) => {
    const isEmailValid = checkEmailValidity(email);
    const isPasswordValid = checkPasswordValidity(password);
    const isConfirmPasswordValid = checkMatch(password, confirmPassword);

    setCheckValues({
      isEmailValid,
      isPasswordValid,
      isConfirmPasswordValid,
    });

    return isEmailValid && isPasswordValid && isConfirmPasswordValid;
  };

  const preparetoSend = () => {
    if (checkValidForm({ ...values, confirmPassword })) {
      sendForm({ ...values });
    }
  };

  const onFirstNameChange = (e) => {
    const firstName = e.target.value;

    setIsFinished(checkIsFinished({ ...values, firstName, confirmPassword }));
    setValues({ ...values, firstName });
  };

  const onLastNameChange = (e) => {
    const lastName = e.target.value;

    setIsFinished(checkIsFinished({ ...values, lastName, confirmPassword }));
    setValues({ ...values, lastName });
  };
  const onEmailChange = (e) => {
    const email = e.target.value;

    setIsFinished(checkIsFinished({ ...values, email, confirmPassword }));
    resetError();
    setValues({ ...values, email });
  };
  const onPasswordChange = (e) => {
    const password = e.target.value;

    setIsFinished(checkIsFinished({ ...values, password, confirmPassword }));
    setValues({ ...values, password: e.target.value });
  };
  const onConfirmPasswordChange = (e) => {
    const confirmPassword = e.target.value;

    setIsFinished(checkIsFinished({ ...values, confirmPassword }));
    setConfirmPassword(confirmPassword);
  };
  const onVisibilityChange = () => setShowPassword(!showPassword);

  const passwordError =
    !checkValues.isConfirmPasswordValid || !checkValues.isPasswordValid;

  const isEmailWrong = isServerError || !checkValues.isEmailValid;
  const emailMessage = isServerError ? errorMessage : 'Email is invalid';

  return (
    <Box>
      <StandartedTextField
        value={values.firstName}
        labelText="First name"
        onChange={onFirstNameChange}
      />
      <StandartedTextField
        labelText="Last name"
        value={values.lastName}
        onChange={onLastNameChange}
      />
      <StandartedTextField
        labelText="Email"
        value={values.email}
        onChange={onEmailChange}
        error={isEmailWrong}
        helperText={isEmailWrong ? emailMessage : null}
      />
      <PasswordForm
        password={values.password}
        onChange={onPasswordChange}
        isError={passwordError}
        errorMessage={
          !checkValues.isConfirmPasswordValid
            ? "Passwords don't match"
            : 'Should contain upper case letters, special symbols, digits and be at least 8 symbols in length.'
        }
        hidden={!showPassword}
        onHiddenChange={onVisibilityChange}
      />
      <PasswordForm
        label="Confirm password"
        password={confirmPassword}
        onChange={onConfirmPasswordChange}
        isError={!checkValues.isConfirmPasswordValid}
        errorMessage="Passwords don't match"
        hidden={!showPassword}
        onHiddenChange={onVisibilityChange}
      />
      <Box sx={{ mt: 4 }} />
      {isLoading ? (
        <LoadingButton label="Loading..." isFullWidth={true} />
      ) : (
        <Button
          fullWidth
          variant="contained"
          onClick={preparetoSend}
          disabled={!isFinished}
        >
          Sign up
        </Button>
      )}
    </Box>
  );
};

Registration.defaultProps = {
  status: IDLE,
  errorMessage: '',
};

Registration.propTypes = {
  sendForm: PropTypes.func.isRequired,
  status: PropTypes.oneOf([PENDING, FULFILLED, REJECTED, IDLE]),
  errorMessage: PropTypes.string,
};

export default Registration;
