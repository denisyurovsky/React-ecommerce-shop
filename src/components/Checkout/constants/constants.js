export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
