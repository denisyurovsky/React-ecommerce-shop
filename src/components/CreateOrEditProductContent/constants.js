export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
