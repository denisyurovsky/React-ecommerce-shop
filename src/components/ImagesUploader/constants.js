export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
