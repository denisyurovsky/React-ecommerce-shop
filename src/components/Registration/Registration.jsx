import { Visibility, VisibilityOff } from '@mui/icons-material';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import FormControl from '@mui/material/FormControl';
import IconButton from '@mui/material/IconButton';
import Input from '@mui/material/Input';
import InputAdornment from '@mui/material/InputAdornment';
import InputLabel from '@mui/material/InputLabel';
import TextField from '@mui/material/TextField';
import React, { useState } from 'react';

import isEmailValid from '../../helpers/isEmailValid';
import isPasswordValid from '../../helpers/isPasswordValid';

import styles from './Registration.module.scss';

const Registration = () => {
  const [values, setValues] = useState({
    showPassword: false,
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
    isEmailValid: true,
    isPasswordValid: true,
    isConfirmPasswordValid: true,
    isFirstNameValid: true,
    isLastNameValid: true,
  });

  const validateForm = (
    email,
    password,
    confirmPassword,
    firstName,
    lastName
  ) => {
    const isFirstNameOk = firstName.length > 0;
    const isLastNameOk = lastName.length > 0;
    const isEmailOk = isEmailValid(email);
    const isPasswordOk = isPasswordValid(password);
    const isConfirmPasswordOk = password === confirmPassword;

    return setValues({
      ...values,
      isEmailValid: isEmailOk,
      isPasswordValid: isPasswordOk,
      isConfirmPasswordValid: isConfirmPasswordOk,
      isFirstNameValid: isFirstNameOk,
      isLastNameValid: isLastNameOk,
    });
  };

  const handleClickShowPassword = () => {
    setValues({
      ...values,
      showPassword: !values.showPassword,
    });
  };

  const onFirstNameChange = (e) => {
    setValues({ ...values, firstName: e.target.value });
  };
  const onLastNameChange = (e) => {
    setValues({ ...values, lastName: e.target.value });
  };
  const onEmailChange = (e) => {
    setValues({ ...values, email: e.target.value });
  };
  const onPasswordChange = (e) => {
    setValues({ ...values, password: e.target.value });
  };
  const onConfirmPasswordChange = (e) => {
    setValues({ ...values, confirmPassword: e.target.value });
  };

  return (
    <Container
      className={styles.containerWrapper}
      maxWidth="sm"
      sx={{ m: 'auto' }}
    >
      <div className={styles.registrationContainer}>
        <TextField
          data-testid="firstNameInput"
          error={!values.isFirstNameValid}
          onChange={onFirstNameChange}
          label="First name"
          variant="standard"
        />
        <TextField
          data-testid="lastNameInput"
          error={!values.isLastNameValid}
          onChange={onLastNameChange}
          label="Last name"
          variant="standard"
        />
        <TextField
          data-testid="emailInput"
          onChange={onEmailChange}
          error={!values.isEmailValid}
          label="Email"
          variant="standard"
        />
        <FormControl variant="standard">
          <InputLabel
            error={!values.isPasswordValid}
            htmlFor="standard-adornment-password"
          >
            Password
          </InputLabel>
          <Input
            data-testid="passwordInput"
            type={values.showPassword ? 'text' : 'password'}
            onChange={onPasswordChange}
            error={!values.isPasswordValid}
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  data-testid="hiddenButton"
                  aria-label="toggle password visibility"
                  onClick={handleClickShowPassword}
                >
                  {values.showPassword ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            }
          />
        </FormControl>
        <FormControl variant="standard">
          <InputLabel
            error={!values.isConfirmPasswordValid}
            htmlFor="standard-adornment-password"
          >
            Confirm Password
          </InputLabel>
          <Input
            data-testid="confirmPasswordInput"
            onChange={onConfirmPasswordChange}
            error={!values.isConfirmPasswordValid}
            type={values.showPassword ? 'text' : 'password'}
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  aria-label="toggle password visibility"
                  onClick={handleClickShowPassword}
                >
                  {values.showPassword ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            }
          />
        </FormControl>
        <Button
          data-testid="signUpButton"
          type="submit"
          fullWidth
          variant="contained"
          sx={{ mt: 3, mb: 2 }}
          onClick={() =>
            validateForm(
              values.email,
              values.password,
              values.confirmPassword,
              values.firstName,
              values.lastName
            )
          }
        >
          Sign Up
        </Button>
      </div>
    </Container>
  );
};

export default Registration;
