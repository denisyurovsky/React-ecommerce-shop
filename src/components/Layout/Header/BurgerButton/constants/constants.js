import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
