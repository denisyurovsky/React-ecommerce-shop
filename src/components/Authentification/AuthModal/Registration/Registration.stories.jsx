import React from 'react';

import Registration from './Registration';

export default {
  title: 'AuthModal/Registration',
  component: Registration,
  argTypes: { sendForm: { action: 'Sending form' } },
};

const Template = (args) => <Registration {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Error = Template.bind({});
Error.storyName = 'Server error';
Error.args = {
  status: 'failed',
  errorMessage: 'User already exists',
};

export const Loading = Template.bind({});
Loading.args = {
  status: 'loading',
};
