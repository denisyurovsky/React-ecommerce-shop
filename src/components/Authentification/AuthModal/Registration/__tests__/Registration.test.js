import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import { authStatus } from '../../../../../constants/authConstants';
import Registration from '../Registration';

const { PENDING, REJECTED } = authStatus;

describe('Registration snapshot', () => {
  it('should render a valid snapshot', () => {
    const { asFragment } = render(<Registration sendForm={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('toggle password input simultaneously', () => {
  beforeEach(() => {
    render(<Registration sendForm={() => {}} />);
  });

  it('should hide both password inputs by default', () => {
    const password = screen.getByLabelText('Password');
    const confPassword = screen.getByLabelText('Confirm password');

    expect(password.getAttribute('type')).toBe('password');
    expect(confPassword.getAttribute('type')).toBe('password');
  });

  describe('show both password inputs', () => {
    const getCheckbox = (index) => {
      const password = screen.getByLabelText('Password');
      const confPassword = screen.getByLabelText('Confirm password');
      const checkbox = screen
        .getAllByTestId('visibility')
        .map((el) => el.querySelector('input'))
        .find((el, i) => i === index);

      return { checkbox, password, confPassword };
    };

    it('should work with first checkbox', () => {
      const { checkbox, password, confPassword } = getCheckbox(0);

      userEvent.click(checkbox);

      expect(password.getAttribute('type')).toBe('text');
      expect(confPassword.getAttribute('type')).toBe('text');
    });

    it('should work with second checkbox', () => {
      const { checkbox, password, confPassword } = getCheckbox(1);

      userEvent.click(checkbox);

      expect(password.getAttribute('type')).toBe('text');
      expect(confPassword.getAttribute('type')).toBe('text');
    });
  });
});

describe('Successful registration scenario', () => {
  it('should log success message', () => {
    const callback = jest.fn();
    const { getByLabelText, getByRole } = render(
      <Registration sendForm={callback} />
    );

    userEvent.type(getByLabelText('First name'), 'Denis');
    userEvent.type(getByLabelText('Last name'), 'Iurovskii');
    userEvent.type(getByLabelText('Email'), 'denis@epam.com');
    userEvent.type(getByLabelText('Password'), 'AbcDefG12!');
    userEvent.type(getByLabelText('Confirm password'), 'AbcDefG12!');
    userEvent.click(getByRole('button', { name: /sign up/i }));

    expect(callback).toHaveBeenCalledTimes(1);
  });

  it('should display loading button', () => {
    const { getByRole } = render(
      <Registration sendForm={() => {}} status={PENDING} />
    );

    expect(getByRole('button', { name: /loading/i })).toBeInTheDocument();
  });
});

describe('Errors in password validation', () => {
  const fillInputs = () => {
    const passwordInput = screen.getByLabelText('Password');
    const confirmPasswordInput = screen.getByLabelText('Confirm password');

    userEvent.type(screen.getByLabelText('First name'), 'Denis');
    userEvent.type(screen.getByLabelText('Last name'), 'Iurovskii');
    userEvent.type(screen.getByLabelText('Email'), 'denis@epam.com');

    return { passwordInput, confirmPasswordInput };
  };

  beforeEach(() => {
    render(<Registration sendForm={() => {}} />);
  });

  it('should be same passwords', () => {
    const { passwordInput, confirmPasswordInput } = fillInputs();

    userEvent.type(passwordInput, 'AbcDefG12');
    userEvent.type(confirmPasswordInput, 'Mssdasgh1!');
    userEvent.click(screen.getByRole('button', { name: /sign up/i }));

    expect(passwordInput.getAttribute('aria-invalid')).toBe('true');
    expect(confirmPasswordInput.getAttribute('aria-invalid')).toBe('true');
    expect(screen.queryAllByText("Passwords don't match")).toHaveLength(2);
  });

  it('should show simple error message', () => {
    const { passwordInput, confirmPasswordInput } = fillInputs();

    userEvent.type(passwordInput, 'AbcDefG12');
    userEvent.type(confirmPasswordInput, 'AbcDefG12');
    userEvent.click(screen.getByRole('button', { name: /sign up/i }));

    expect(
      screen.getByText(
        'Should contain upper case letters, special symbols, digits and be at least 8 symbols in length.'
      )
    ).toBeInTheDocument();
  });
});

describe('Email validation', () => {
  it('should display server error', () => {
    const serverError = 'Testin server error';
    const { getByText } = render(
      <Registration
        sendForm={() => {}}
        status={REJECTED}
        errorMessage={serverError}
      />
    );

    expect(getByText(serverError)).toBeInTheDocument();
  });

  it('should display email validation error', () => {
    const { getByLabelText, getByRole } = render(
      <Registration sendForm={() => {}} />
    );

    userEvent.type(getByLabelText('First name'), 'Denis');
    userEvent.type(getByLabelText('Last name'), 'Iurovskii');
    userEvent.type(getByLabelText('Email'), 'denis@epam');
    userEvent.type(getByLabelText('Password'), 'Mssdasgh1!');
    userEvent.type(getByLabelText('Confirm password'), 'Mssdasgh1!');
    userEvent.click(getByRole('button', { name: /sign up/i }));

    expect(screen.getByText('Email is invalid')).toBeInTheDocument();
  });
});
