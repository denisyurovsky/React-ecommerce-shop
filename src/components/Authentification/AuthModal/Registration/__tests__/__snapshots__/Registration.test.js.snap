// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration snapshot should render a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-1"
        id="mui-1-label"
      >
        First name
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-1"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-2"
        id="mui-2-label"
      >
        Last name
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-2"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-3"
        id="mui-3-label"
      >
        Email
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-3"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-4"
        id="mui-4-label"
      >
        Password
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-ghsjzk-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-4"
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="visibility"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityOffIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-5"
        id="mui-5-label"
      >
        Confirm password
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-ghsjzk-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="mui-5"
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="visibility"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityOffIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-h5fkc8"
    />
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      Sign up
    </button>
  </div>
</DocumentFragment>
`;
